db/migrate/20180208053256_create_user_columns.rb

```diff
@@ -0,0 +1,13 @@
+class CreateUserColumns < ActiveRecord::Migration[5.1]
+  def change
+    create_table :user_columns do |t|
+      t.string :name, :limit => 255, :null => false
```

Comment:
Shouldn't this be `resource` or `controller` or something more descriptive? it took me a bit to understand what the name means.

Reply:
I had it as resource/controller before @tstrachota felt on some pages we'd want to have more than 1 set of table preferences.  So I changed it. I saw 'name' as something more generic (like a key.)  Its entirely upto the UI to figure out what to put there.

Reply:
Yes, I think it's better to use a generic name and don't limit it only to resources or controllers.

Reply:
Maybe `table_name` could work if we want to be a bit more specific.

Reply:
I renamed the model to TablePreferences.  I think that should be enough to indicate name = tablename imo